src/NHibernate/Linq/Functions/BaseHqlGeneratorForMethod.cs

```diff
@@ -7,10 +7,17 @@
 
 namespace NHibernate.Linq.Functions
 {
-    public abstract class BaseHqlGeneratorForMethod : IHqlGeneratorForMethod
-    {
-        public IEnumerable<MethodInfo> SupportedMethods { get; protected set; }
+	public abstract class BaseHqlGeneratorForMethod : IHqlGeneratorForMethod, IHqlGeneratorForMethodExtended
+	{
+		public IEnumerable<MethodInfo> SupportedMethods { get; protected set; }
 
-        public abstract HqlTreeNode BuildHql(MethodInfo method, Expression targetObject, ReadOnlyCollection<Expression> arguments, HqlTreeBuilder treeBuilder, IHqlExpressionVisitor visitor);
-    }
-}
\ No newline at end of file
+		public abstract HqlTreeNode BuildHql(MethodInfo method, Expression targetObject, ReadOnlyCollection<Expression> arguments, HqlTreeBuilder treeBuilder, IHqlExpressionVisitor visitor);
+
+		public virtual bool AllowsNullableReturnType(MethodInfo method) => true;
+
+		bool IHqlGeneratorForMethodExtended.AllowsNullableReturnType(MethodInfo method)
```

Comment:
I do not get why an explicit implementation is needed here.

Reply:
Removed as it is not needed.